### PR TITLE
[node][static-build][redwood] add root path pattern to prepareCache

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
### Related Issues

Adds root path caching support to other builders

#### Tests

- [ ] The code changed/added as part of this PR has been covered with tests
- [ ] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/node/src/index.ts [hunk 1]
```diff
@@ -381,7 +381,10 @@ export const build: BuildV3 = async ({
 };
 
 export const prepareCache: PrepareCache = async ({ workPath }) => {
-  const cache = await glob('node_modules/**', workPath);
+  const cache = {
```
Comment (Ethan-Arrowood) [inline packages/node/src/index.ts]: @TooTallNate @styfle - i didn't see a test for this block. Any ideas how I can test it? This package has an integration test folder but those don't assert the cache as far as i can tell. WDYT?

File: packages/static-build/src/index.ts [hunk 2]
```diff
@@ -786,6 +786,8 @@ export const prepareCache: PrepareCache = async ({
     await glob('{.shadow-cljs,node_modules}/**', workPath)
   );
 
+  Object.assign(cacheFiles, await glob('**/node_modules/**', workPath));
```
Comment (styfle) [inline packages/static-build/src/index.ts]: Is there a way to get the glob pattern above "Default cache files" to include this pattern as well? That way we don't need to pass over all the files twice?

Comment (Ethan-Arrowood) [reply packages/static-build/src/index.ts]: let me try adjusting the previous one - i don't see why it wouldn't include all the paths

File: packages/node/src/index.ts [hunk 3]
```diff
@@ -381,7 +381,10 @@ export const build: BuildV3 = async ({
...
+    ...(await glob('node_modules/**', workPath)),
+    ...(await glob('**/node_modules/**', workPath)),
```
Comment (styfle) [inline packages/node/src/index.ts]: Same for this glob, can it be a single pattern to match both cases?

File: packages/node/src/index.ts (hunk 1 above)
Comment (styfle) [reply packages/node/src/index.ts]: I think the same style of unit tests for static-build are best

File: packages/node/src/index.ts (hunk 3 above)
Comment (TooTallNate) [reply packages/node/src/index.ts]: I think we need to consider `repoRootPath` here, when it's set.

File: packages/node/src/index.ts (hunk 1 above)
Comment (TooTallNate) [reply packages/node/src/index.ts]: I suggest taking inspiration from [here](https://github.com/vercel/vercel/blob/main/packages/static-build/test/prepare-cache.test.ts) (that is, unit test the `prepareCache()` function).

**Review by TooTallNate — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Comment by Ethan-Arrowood** [comment]
Any ideas why the E2E tests keep hanging indefinitely? @styfle @TooTallNate

**Comment by styfle** [comment]
@Ethan-Arrowood Looks like its not hanging anymore but one did time out. You can also run E2E tests locally to confirm they work since GH Actions has been really flaky with network IO.

**Comment by Ethan-Arrowood** [comment]
i will attempt to run locally - usually many things break when I run the whole suite locally

**Pull request merged**